File: 2024/Boj_10773.cpp
#include <iostream>
#include <stack>

using namespace std;

int main(){
    int k;
   
    stack<int> stack;
    
    cin >> k;

   for(int i; i<k; i++){
        int num;
        cin >> num;

        if(num==0){
            stack.pop();
        } else {
            stack.push(num);
        }
   }
   int sum = 0;

   while(!stack.empty()){
    sum += stack.top();
    stack.pop();
   }

   cout << sum;
}

File: 2025/2025-java/2025-java(new)/src/Boj_10773.java
import java.util.Scanner;
import java.util.Stack;

public class Boj_10773 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Stack<Integer> stack = new Stack<>();
        int n = scanner.nextInt();
        int sum = 0;

        for (int i = 0; i < n; i++) {
            int num = scanner.nextInt();

            if (num == 0) {
                stack.pop();
            } else {
                stack.push(num);
            }
        }
        while (!stack.isEmpty()) {
            sum += stack.peek();
            stack.pop();
        }
        System.out.println(sum);
    }
}